test/com/ableton/JenkinsMocks.groovy

```diff
@@ -9,4 +9,12 @@ class JenkinsMocks {
   static Closure isUnix = {
     return !System.properties['os.name'].toLowerCase().contains('windows')
   }
+
+  static Closure pwd = { args ->
+    if (args?.tmp ?: false) {
```

Comment:
I'm not sure I know groovy enough to know what ?. and ?: mean. Does this mean if args.tmp is false? If so, why do we return 'java.io.tmpdir' here?

Reply:
`a?.b` is the short version for `a == null ? null : a.b`  and `a ?: b` is the short version for `a ? a : b`.

Reply:
@mst-ableton In case you want to read more, `?.` is formally called the [safe navigation operator](http://groovy-lang.org/operators.html#_safe_navigation_operator) and `?:` is called the [elvis operator](http://groovy-lang.org/operators.html#_elvis_operator).

Reply:
Thanks for the clarification. In that case I'm not sure why the `?:` is needed here. Let's say `q = args?.tmp`. Then we're doing `if (q ?: false) {`.  In that case, if `q` evaluates to `true` the elvis operator will return `q` (which, again, evaluates to `true`) and if `q` evaluates to false the elvis operator will return `false`. Since the results go directly into an if statement, I'm not sure what extra the elvis operator accomplishes.

Reply:
I guess you're right, I'll simplify this statement.